sqlmesh/core/plan/evaluator.py

```diff
@@ -159,17 +159,20 @@ def _restate(self, plan: Plan) -> None:
         if not plan.restatements:
             return
 
-        all_snapshots = (
-            [s for s in plan.snapshots if s.name in plan.restatements]
-            if plan.is_dev
-            else self.state_sync.get_snapshots_by_models(*plan.restatements)
-        )
-        self.state_sync.remove_interval(
-            [],
-            start=plan.start,
-            end=plan.end,
-            all_snapshots=all_snapshots,
-        )
+        target_snapshots = [s for s in plan.snapshots if s.name in plan.restatements]
```

Comment:
what's going on here?

Reply:
Discussed in person. This is to make sure that when a user triggers restatement in the development environment, it only impacts snapshots in that target environment, without impacting other environments if possible.